src/steamship/agents/schema/action.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any, List, Optional, Union
+
+from pydantic import BaseModel
+
+from steamship import Block, Task
+from steamship.agents.schema.tool import AgentContext, Tool
+
+
+class Action(BaseModel):
+    """Actions represent a binding of a Tool to the inputs supplied to the tool.
+
+    Upon completion, the Action also contains the output of the Tool given the inputs.
+    """
+
+    tool: Tool
+    """Tool used by this action."""
+
+    input: List[Block]
+    """Data provided directly to the Tool (full context is passed in run)."""
+
+    output: Optional[List[Block]] = []
+    """Any direct output produced by the Tool."""
+
+
+class AgentTool(Tool):
```

Comment:
nit/readability - I was confused by `AgentTool` as a name.

I think the idea is that we want `Tool` to be an ABC, but we also want something that suffices as a short-cut for `NoOp`?

Maybe we just call it `NoOpTool` (if that's what it is)?

Reply:
or FinishTool